homeassistant/components/ezviz/config_flow.py

```diff
@@ -129,12 +132,45 @@ async def async_step_user(self, user_input=None):
             {
                 vol.Required(CONF_USERNAME): str,
                 vol.Required(CONF_PASSWORD): str,
-                vol.Optional(CONF_REGION, default=DEFAULT_REGION): str,
+                vol.Required(CONF_REGION, default=DEFAULT_REGION): vol.In(
+                    [DEFAULT_REGION, RUSSIA_REGION, CONF_CUSTOMIZE]
+                ),
             }
         )
 
         return self.async_show_form(
-            step_id="user", data_schema=data_schema, errors=errors or {}
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_user_custom_region_url(self, user_input):
+        """Handle a flow initiated by the user for custom region url."""
+
+        errors = {}
+
+        if user_input[CONF_REGION] != CONF_CUSTOMIZE:
+            if CONF_TIMEOUT not in user_input:
+                user_input[CONF_TIMEOUT] = DEFAULT_TIMEOUT
+
+            try:
+                return await self._validate_and_create_auth(user_input)
+
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+        data_schema_custom_url = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=user_input[CONF_USERNAME]): str,
+                vol.Required(CONF_PASSWORD, default=user_input[CONF_PASSWORD]): str,
+                vol.Required(CONF_REGION, default=DEFAULT_REGION): str,
```

Comment:
Shouldn't this be url instead of region?

Reply:
Done. Changed all region references to URL.